recipes-ni/initscripts-nilrt/files/nisetserialnumber

```diff
@@ -0,0 +1,12 @@
+#!/bin/bash
+
+# If the serial number is undefined in hardware, create one from a MAC address.
+
+. /etc/natinst/networking/functions.common
+
+SERIAL=`/sbin/fw_printenv -n serial# 2>&1`
+if [ "$SERIAL" = "ABCDEFG" ]; then
```

Comment:
I wonder if this is still needed. I can't find any evidence that fw_printenv will return this when the serial# is not set.
I've checked /sbin/fw_printenv and /usr/share/nisysinfo/smbios_helper. Also checked a qemu VM where the serial number is not set when running 'dmidecode -t 1' and in that case 'fw_printenv -n serial#' returns '00123456' not 'ABCDEFG'

I think  we can probably drop this hack.

Reply:
The ABCDEFG probably came from [here](https://github.com/ni/meta-orgconf/blob/nilrt/master/dunfell/recipes-core/images/safemode-image/grubenv_non_ni_target#L2). We don't use this file in current image but I wonder if we should.
Serial number seems to be used to set hostname. If we drop this, wouldn't all VMs on network having the same hostname create problems?

Reply:
I think we'd like to keep this behavior of generating a unique hostname.

Reply:
Ok. Keeping it then